torchtext/data/functional.py

```diff
@@ -39,20 +39,26 @@ def generate_sp_model(filename, vocab_size=20000,
     torch.ops.torchtext.generate_sp_model(filename, vocab_size, model_type, model_prefix)
 
 
-def load_sp_model(spm_path):
+def load_sp_model(spm):
     r"""Load a  sentencepiece model for file.
 
     Arguments:
-        spm_path: the file path saving the sentencepiece model.
+        spm: the file path or a file object saving the sentencepiece model.
 
     Outputs:
         output: a SentencePiece model.
 
     Examples:
         >>> from torchtext.data.functional import load_sp_model
         >>> sp_model = load_sp_model("m_user.model")
+        >>> sp_model = load_sp_model(open("m_user.model", 'rb'))
     """
-    return torch.ops.torchtext.load_sp_model(spm_path)
+    if isinstance(spm, str):
+        return torch.ops.torchtext.load_sp_model(spm)
+    elif isinstance(spm, io.BufferedReader):
+        return torch.ops.torchtext.load_sp_model_string(spm.read())
+    else:
+        raise RuntimeError('the input of the load_sp_model func is not supported.')
```

Comment:
@zhangguanheng66  PathManager.open() returns "_io.TextIOWrapper" type and will fail here. How about remove the check?
```suggestion
    return torch.ops.torchtext.load_sp_model_string(spm.read())
```

Reply:
I notice that and was thinking about adding that check. 
However, with `PathManager.open() `, it will return f as `_io.TextIOWrapper`. It seems not working with sentencepiece constructor. It needs to be binary reading where  `PathManager.open(, 'rb')` returns `io.BufferedReader`.

Reply:
@zhangguanheng66 you are right! It LGTM. Note: there is a test failure